File: modulesBase/libBase/src/main/java/com/example/libbase/base/app/BaseApplication.java
package com.example.libbase.base.app;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;
import cn.pivotstudio.husthole.moduleb.network.NetworkApi;
import cn.pivotstudio.husthole.moduleb.network.NetworkRequiredInfo;
import cn.pivotstudio.moduleb.database.AppDatabase;
import com.alibaba.android.arouter.launcher.ARouter;
import com.example.libbase.BuildConfig;
import com.example.libbase.base.ui.activity.ActivityManager;
import com.example.libbase.util.emoji.EmotionUtil;

/**
 * @classname:BaseApplication
 * @description:
 * @date:2022/4/28 19:55
 * @version:1.0
 * @author:
 */

public class BaseApplication extends Application {

    @SuppressLint("StaticFieldLeak")
    public static Context context;

    //数据库
    public static AppDatabase db;

    @Override
    public void onCreate() {
        super.onCreate();
        context = getApplicationContext();
        //初始化网络框架
        NetworkApi.init(new NetworkRequiredInfo(this));
        //创建本地数据库
        db = AppDatabase.getInstance(this);

        if (!BuildConfig.isRelease) {
            ARouter.openLog();     // 打印日志
            ARouter.openDebug();
        }
        //组件化初始化
        ARouter.init(this);

        //表情包资源初始化
        EmotionUtil.init(this);
    }

    public static Context getContext() {
        return context;
    }

    public static AppDatabase getDb() {
        return db;
    }

    public static ActivityManager getActivityManager() {
        return ActivityManager.getInstance();
    }
}

File: modulesBase/libBase/src/main/java/cn/pivotstudio/moduleb/libbase/base/app/BaseApplication.kt
package cn.pivotstudio.moduleb.libbase.base.app

import cn.pivotstudio.moduleb.database.HustHoleRoomDatabase
import com.alibaba.android.arouter.launcher.ARouter
import cn.pivotstudio.moduleb.libbase.util.emoji.EmotionUtil
import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import cn.pivotstudio.husthole.moduleb.network.*
import cn.pivotstudio.moduleb.libbase.BuildConfig
import cn.pivotstudio.moduleb.libbase.base.ui.activity.ActivityManager

/**
 * @classname:BaseApplication
 * @description:
 * @date:2022/4/28 19:55
 * @version:1.0
 * @author:
 */
class BaseApplication : Application() {

    init {
        NetworkApi.init(NetworkRequiredInfo(this))
        HustHoleApi.init(this)
    }

    val database: HustHoleRoomDatabase by lazy { HustHoleRoomDatabase.getInstance(this) }

    override fun onCreate() {
        super.onCreate()
        context = applicationContext
        DB = database

        //创建本地数据库
        if (!BuildConfig.isRelease) {
            ARouter.openLog() // 打印日志
            ARouter.openDebug()
        }
        //组件化初始化
        ARouter.init(this)

        //表情包资源初始化
        EmotionUtil.init(this)
    }

    companion object {
        @JvmField
        @SuppressLint("StaticFieldLeak")
        var context: Context? = null

        //数据库
        @Deprecated("Java遗留")
        @JvmStatic
        var DB: HustHoleRoomDatabase? = null

        @JvmStatic
        val activityManager: ActivityManager
            get() = ActivityManager.getInstance()
    }
}